storages/storages.go

```diff
@@ -79,7 +80,12 @@ func (b *Backend) RoundTrip(r *http.Request) (*http.Response, error) {
 			origErr: fmt.Errorf("backend %v in maintenance mode", b.Name)}
 	}
 	err := error(nil)
-	resp, oerror := b.RoundTripper.RoundTrip(r)
+
+	transportName := b.Transports.SelectTransport(r.Method, r.URL.Path, r.URL.RawQuery)
```

Comment:
this should be encapsulated in Transports.RoundTripp method and as Lukasz suggested put transports into roundtripper field